gobblin-modules/gobblin-orc/src/main/java/org/apache/gobblin/writer/GobblinBaseOrcWriter.java

```diff
@@ -258,7 +261,18 @@ public void close()
   public void commit()
       throws IOException {
     closeInternal();
+    if(this.validateORCDuringCommit) {
+      try {
+        OrcFile.createReader(this.stagingFile, new OrcFile.ReaderOptions(conf));
+      } catch (Exception e) {
+        log.error("Found error when validating ORC file during commit phase", e);
+        HadoopUtils.deletePath(this.fs, this.stagingFile, false);
```

Comment:
Do we want to do this validation here or if its being published at a folder level at a different step, is when we have the reader validate all the files going into publish? My concern is that even if we are validating and deleting files at the writer step, this still assumes that this writer will shut down using a happy path and not suddenly. If it shuts down due to some container error code e.g. 143 and not go through the cleanup process then it can still be moved to another folder later down the line?